### DISCUSSION TITLE
How to construct a multiline textbox field

### DISCUSSION THREAD

**Discussion by mavam** [ArthurSonzogni/FTXUI#543] [Q&A] [answered]
I was wondering what it takes to create a multiline version of `Input`. I was initially thinking of a vertical container of multiple `Input` instances, but then the overflow semantics would have to be manually added on top, which seems easier to do at a lower level, i.e., at the implementation of `Input` itself. But maybe there's an easy trick to sync the viewable window of all input lines in a container?

**Comment by ArthurSonzogni** [comment, marked as answer]
There are no such a component. We would have to work on it. I agree this would be better addressed as a built-in one provided by the library.

**Reply by mavam** [reply]
Okay, once this becomes a top priority I'll dig deeper into the implementation of `Input` to come up with a strategy.

**Reply by ArthurSonzogni** [reply]
I started prototyping it on the `textarea` branch:
https://github.com/ArthurSonzogni/FTXUI/tree/textarea

**Reply by ArthurSonzogni** [reply]
Merged: https://github.com/ArthurSonzogni/FTXUI/commit/7de4f8683d530ff5573483f40b17df4b3c6ed736

**Reply by mavam** [reply]
Saweeeet! I hope I can give it spin soon. I have a multi-part blog post about FTXUI and my efforts in the pipeline, but it's still going to take me a bit to publish it and still a cool story.
